fb-negotiation/core/kb.py

```diff
@@ -0,0 +1,36 @@
+from cocoa.core.kb import KB as BaseKB
+
+class KB(BaseKB):
+    def __init__(self, attributes, facts):
+        super(KB, self).__init__(attributes)
+        self.facts = facts
+        # "Attributes" is the schema of attributes
+        # Facts are the instiation of those atrributes for a certain agent
+
+    def to_dict(self):
+        return self.facts
+
+    @staticmethod
+    def from_dict(attributes, raw):
+        return KB(attributes, raw)
+
+    def dump(self):
+        role = self.facts['Role']
+        print 'Role: {} speaker'.format(role)
+
+        books, hats, balls = self.facts['Item_counts'][0], self.facts['Item_counts'][1], self.facts['Item_counts'][2]
```

Comment:
Can we make item_counts and item_values a dict, e.g. {'book': 1, 'hat': 1, 'ball':2}